core/core.go

```diff
@@ -354,6 +370,10 @@ func (c *Core) WriteBlock(block *types.Block) {
 		nodeCtx := common.NodeLocation.Context()
 		if order == nodeCtx {
 			c.addToAppendQueue(block)
+			parentHeader := c.GetHeader(block.ParentHash(), block.NumberU64()-1)
+			if parentHeader != nil {
```

Comment:
If parent is nil we should log an error, no? And send to missing parent feed

Reply:
After discussing, the parent being nil from GetHeader() likely means that the parent is in the append queue, which is not necessarily an error